Fix counter collection inconsistency with rocprofv3

feizheng10 opened the pull request:
For some case, we noticed collected total row numbers are not the same across all perfmon csv files:
pmc_perf.csv, SQ_IFETCH_LEVEL.csv, SQ_INST_LEVEL_LDS.csv, SQ_INST_LEVEL_SMEM.csv, SQ_INST_LEVEL_VMEM.csv.
The quick solution is: inner join if collected csv are not on the same page.
We might need to dig into and check the root reason.

feizheng10 committed:
inner join if collected csv are not on the same page

feizheng10 committed:
format code

feizheng10 committed:
fix potential rocprofv3 env path hack

vedithal-amd commented on src/rocprof_compute_soc/soc_base.py:
```diff
@@ -304,9 +304,9 @@ def add(self, counter) -> bool:
+# FIXME: This is a HACK
 def using_v3():
-    return "ROCPROF" in os.environ.keys() and os.environ["ROCPROF"] == "rocprofv3"
+    return "ROCPROF" in os.environ.keys() and "rocprofv3" in os.environ["ROCPROF"]
```
Just curious as to why does this hack do substring match?

vedithal-amd replied:
I get it now, ROCPROF env. var. will contain the path to rocprofv3, in that case should we use `os.environ["ROCPROF"].endswith("rocprofv3")`

feizheng10 replied:
[bc9cf07](https://github.com/ROCm/rocprofiler-compute/pull/589/commits/bc9cf07dc93392364b8aa7108f1fc2769f80542c)

vedithal-amd committed:
Fix post analysis gui in standalone binary (#591)

* Fix post analysis gui in standalone binary

* Add post analysis gui assets and required server libraries for GUI
  server and web page

* Add port forwarding to docker test compose

* Update README me to use `docker compose up` instead of `docker compose run`
  to run containers with port forwarding and to leverage other
  functionalities of docker compose

xuchen-amd committed:
Fix rocprofv1 output processing. (#588)

feizheng10 committed:
Merge branch 'develop' into fix_collection_inconsistency

feizheng10 committed:
Merge branch 'develop' into fix_collection_inconsistency

vedithal-amd commented on src/utils/file_io.py:
```diff
@@ -209,7 +209,8 @@ def create_single_df_pmc(raw_data_dir, node_name, kernel_verbose, verbose):
 
-        final_df = pd.concat(dfs, keys=coll_levels, axis=1, copy=False)
+        # TODO: double check the case if all tmp_df.shape[0] are not on the same page
+        final_df = pd.concat(dfs, keys=coll_levels, axis=1, join="inner", copy=False)
```
As discussed in the meeting, inner join based on index is the only option we have since kernel name and dispatch ids might be non-deterministic

vedithal-amd reviewed (approved):
LGTM, just some minor comments

feizheng10 committed:
improve detecting rocprofv3

feizheng10 committed:
format code
